demos/smart_classroom_demo_gapi/include/initialize.hpp

```diff
@@ -0,0 +1,375 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "custom_kernels.hpp"
+#include "smart_classroom_demo.hpp"
+#include "actions.hpp"
+#include "kernel_packages.hpp"
+
+#include <opencv2/gapi/infer/ie.hpp>
+#include <opencv2/gapi/infer/parsers.hpp>
+
+namespace nets {
+    G_API_NET(FaceDetector,        <cv::GMat(cv::GMat)>, "face-detector");
+    G_API_NET(LandmarksDetector,   <cv::GMat(cv::GMat)>, "landmarks-detector");
+    G_API_NET(FaceReidentificator, <cv::GMat(cv::GMat)>, "face-reidentificator");
+    using PAInfo = std::tuple<cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat, cv::GMat>;
+    G_API_NET(PersonDetActionRec,  <PAInfo(cv::GMat)>,   "person-detection-action-recognition");
+} // namespace nets
+
+namespace util {
+bool ParseAndCheckCommandLine(int argc, char *argv[]) {
+    gflags::ParseCommandLineNonHelpFlags(&argc, &argv, true);
+    if (FLAGS_h) {
+        showUsage();
+        showAvailableDevices();
+        return false;
+    }
+    slog::info << "Parsing input parameters" << slog::endl;
+    if (FLAGS_i.empty()) {
+        throw std::logic_error("Parameter -i is not set");
+    }
+    if (FLAGS_m_act.empty() && FLAGS_m_fd.empty()) {
+        throw std::logic_error("At least one parameter -m_act or -m_fd must be set");
+    }
+    return true;
+}
+} // namespace util
+
+namespace config {
+const std::array<std::string, 7> action_detector_5 = {
+    "mbox_loc1/out/conv/flat",
+    "mbox_main_conf/out/conv/flat/softmax/flat",
+    "mbox/priorbox",
+    "out/anchor1",
+    "out/anchor2",
+    "out/anchor3",
+    "out/anchor4"};
+
+const std::array<std::string, 7> action_detector_6 = {
+    "ActionNet/out_detection_loc",
+    "ActionNet/out_detection_conf",
+    "ActionNet/action_heads/out_head_1_anchor_1",
+    "ActionNet/action_heads/out_head_2_anchor_1",
+    "ActionNet/action_heads/out_head_2_anchor_2",
+    "ActionNet/action_heads/out_head_2_anchor_3",
+    "ActionNet/action_heads/out_head_2_anchor_4"};;
+
+inline char separator() {
+#ifdef _WIN32
+    return '\\';
+#else
+    return '/';
+#endif
+}
+
+bool fileExists(const std::string& name) {
+    std::ifstream f(name.c_str());
+    return f.good();
+}
+
+std::string folderName(const std::string& path) {
+    size_t found_pos;
+    found_pos = path.find_last_of(separator());
+    if (found_pos != std::string::npos)
+        return path.substr(0, found_pos);
+    return std::string(".") + separator();
+}
```

Comment:
I see too much code for .hpp here, and the functions are neither static nor inline

May cause problems if the header is included more than once

Also, don't see a point in having so much code in `.hpp` if it is not a template.